xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/PageNotificationEventDescriptorClass.xml

```diff
@@ -0,0 +1,194 @@
+    <comment/>
+    <minorEdit>false</minorEdit>
+    <syntaxId>xwiki/2.1</syntaxId>
+    <hidden>true</hidden>
+    <content/>
+    <class>
+        <name>XWiki.Notifications.Code.PageNotificationEventDescriptorClass</name>
+        <customClass/>
+        <customMapping/>
+        <defaultViewSheet/>
+        <defaultEditSheet/>
+        <defaultWeb/>
+        <nameField/>
+        <validationScript/>
+        <applicationName>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>applicationName</name>
+            <number>1</number>
+            <picker>0</picker>
+            <prettyName>Application</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </applicationName>
+        <eventType>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>eventType</name>
+            <number>2</number>
+            <picker>0</picker>
+            <prettyName>Event type</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </eventType>
+        <eventPrettyName>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>eventPrettyName</name>
+            <number>3</number>
+            <picker>0</picker>
+            <prettyName>Event name</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </eventPrettyName>
+        <eventIcon>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>eventIcon</name>
+            <number>4</number>
+            <picker>0</picker>
+            <prettyName>Event icon</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </eventIcon>
+        <listenTo>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>listenTo</name>
+            <number>5</number>
+            <picker>0</picker>
+            <prettyName>Listen to …</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
```

Comment:
Actually if you use a static list, it must use relational storage, multiple select, and display type: input. I don't know if the input displayer handle the other options.
In the watchlist, we use textarea to avoid the field limits (255 characters for a string).